docs/source/howto/data.rst

```diff
@@ -260,6 +260,338 @@ Finding data
+     25082  StructureData  229D:02h:51m ago
+     41209  StructureData  229D:02h:51m ago
+    119793  StructureData  124D:00h:46m ago
+     36721  StructureData  229D:07h:03m ago
+     12639  StructureData  229D:00h:18m ago
+    129749  StructureData  120D:12h:33m ago
+    105044  StructureData  120D:13h:50m ago
+     55375  StructureData  229D:06h:54m ago
+    103625  StructureData  123D:23h:44m ago
+      1830  StructureData  229D:04h:47m ago
+    113308  StructureData  124D:03h:52m ago
+    101526  StructureData  112D:10h:14m ago
+    100501  StructureData  229D:06h:54m ago
+     63252  StructureData  229D:00h:17m ago
+    112654  StructureData  120D:11h:40m ago
+    115210  StructureData  112D:07h:51m ago
+     70409  StructureData  227D:19h:48m ago
+    103943  StructureData  112D:09h:10m ago
+    104579  StructureData  112D:10h:15m ago
+     73731  StructureData  228D:04h:22m ago
+
+The structures that were added to the group can now be used further.
+
+
+Case II: work with data in the groups
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The data that are organized in the groups can now be accessed separately.
+Let's make a prototype script that would allow us to submit calculations for the structures belonging to the ``promising_structures`` group.
+For that we need to make a query:
+
+.. code-block:: python
+
+    q = QueryBuilder()
+    q.append(Group, filters={'label':'promising_structures'}, tag='group')
+    q.append(StructureData, with_group='group')
+
+The results of this query can be used, for instance, for subsequent simulations:
+
+.. code-block:: python
+
+    for structure in q.all(flat=True):
+        builder = SomeWorkChain.get_builder()
+        builder.structure = structure
+        ...
+        submit(builder)
+
+Note, however, that one can also use ``group.nodes`` to access the nodes of the group.
+To achieve the same result ab above one would need to do something as follows:
+
+.. code-block:: python
+
+    group = Group.get(label='promising_structures')
+    # Here make sure to include only structures, as group can contain any nodes.
+    structures = [s for s in group.nodes if isinstance(nodes, StructureData)]
+    for structure in structures:
+        builder = SomeWorkChain.get_builder()
+        builder.structure = structure
+        ...
+        submit(builder)
+
+
+For the postprocessing of the results, one can also make use of the groups.
+To find the structure that has a property ``property_a`` lower than ``1`` and at the same time belongs to the ``promising_structures`` group, one would need to build a query that looks as follows:
+
+.. code-block:: python
+
+    q = QueryBuilder()
+    q.append(Group, filters={'label':'promising_structures'}, tag='group')
+    q.append(StructureData, with_group='group', tag='structure', project='*')
+    q.append(SomeWorkChain, with_incoming='structure', tag='calculation')
+    q.append(Dict, with_incoming='calculation', filters={'attributes.property_a':{'<':1}})
```

Comment:
```suggestion
    builder = QueryBuilder()
    builder.append(Group, filters={'label': 'promising_structures'}, tag='group')
    builder.append(StructureData, with_group='group', tag='structure', project='*')
    builder.append(SomeWorkChain, with_incoming='structure', tag='calculation')
    builder.append(Dict, with_incoming='calculation', filters={'attributes.property_a': {'<': 1}})
```

Reply:
Do we have some convention for this? I'd imagine that the `QueryBuilder` class is used quite often and maybe calling it `qb` every time is actually clearer than `builder` considering that there are other `builder` instances in the aiida-core namespace?

Reply:
There is no explicit convention for the docs yet, but in the code base, I have been steadily converting all usage to `builder` for consistency, especially in the ORM itself. There is still quite a bit of code that uses all other sort of variables, e.g., `qb`, `qbl`, `qb_object`, `q` and lots of others.

Reply:
I think we should have a brief discussion about this with the team, because I am not convinced that `builder` is the optimal choice here. There are certain cases where using very short, but consistent variable names is preferable, e.g., `np` for `numpy` or `pd` for `pandas` comes to mind.

Reply:
Fair enough. The most important part for me is intelligibility and consistency

Reply:
@yakutovicha let's use `qb` for now, which is also what Marnik is using in other PRs. So at least the documentation for now is consistent

Reply:
heh, I just replaced everything with `builder`

Reply:
replaced `builder` with `qb` in 6d498b7